src/VisualStudio/Core/Def/LanguageService/AbstractPackage`2.cs

```diff
@@ -79,10 +87,10 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
         LoadComponentsInUIContextOnceSolutionFullyLoadedAsync(cancellationToken).Forget();
     }
 
-    protected override async Task LoadComponentsAsync(CancellationToken cancellationToken)
+    protected override Task LoadComponentsAsync(CancellationToken cancellationToken)
     {
-        // Do the MEF loads and initialization in the BG explicitly.
-        await TaskScheduler.Default;
+        // Should only be called from a threadpool thread as this may do MEF loads and initialization
+        Contract.ThrowIfTrue(JoinableTaskFactory.Context.IsOnMainThread);
```

Comment:
This might on UI thread I think.

Reply:
It's on a bg thread. OnAfterPackageLoadedAsync calls LoadComponentsInUIContextOnceSolutionFullyLoadedAsync  which calls this.

Reply:
oh, the OnAfterPackageLoadedAsync call is done with ConfigureAwait(false) -- let me look a bit more.

Reply:
Changed the base.OnAfterPackageLoadedAsync call to capture the calling context, so LoadComponentsInUIContextOnceSolutionFullyLoadedAsync is now always called from a bg thread.